### PR TITLE
refactor: refactor styles modes and feat login and sign up pages

### PR DESCRIPTION
**Pull request by Keretcha**

### CODE REVIEW

File: app/(auth)/logIn/Authorization.module.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+@import '../../Colors/dark-colors.scss';
+@import '../../Colors/light-colors.scss';
+
+.container {
+  background-color: $background-color-light-mode;
+  color: $primary-black;
+  display: flex;
+  justify-content: center;
+  padding-bottom: 48px;
+}
```
Comment (pailodze) [inline app/(auth)/logIn/Authorization.module.scss:11]: add enters after css rule block ends

File: app/(auth)/logIn/page.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+const Authorization: AuthorizationType = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+  const router: AppRouterInstance = useRouter();
+
+  const onSubmit = async (values: object): Promise<void> => {
```
Comment (pailodze) [inline app/(auth)/logIn/page.tsx:26]: values: object ?? passing typescript here?

File: app/(auth)/logIn/page.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          <div className={styles.image}>
+            <Image
+              src={'/icons/test1.svg'}
+              alt={'brand'}
+              width={644}
+              height={575}
+            />
+          </div>
+          <div className={styles.form}>
+            <form
```
Comment (pailodze) [inline app/(auth)/logIn/page.tsx:58]: componentize this form

File: app/(auth)/logIn/page.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
+'use client';
+import axios, { AxiosResponse } from 'axios';
```
Comment (pailodze) [inline app/(auth)/logIn/page.tsx:2]: name page login not logIn there should not be upper case in page name

File: app/(auth)/signUp/page.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  const {
+    register,
+    handleSubmit,
+    setError,
+    watch,
+    formState: { errors },
+  } = useForm();
+
+  const password: string = watch('password');
+  const rePassword: string = watch('rePassword');
```
Comment (pailodze) [inline app/(auth)/signUp/page.tsx:26]: const confirmPassword

File: app/(auth)/signUp/page.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import { ButtonTypeEnum } from '../../Components/Button/enums/button-type.enum';
+import Heading from '../../Components/Heading/Heading';
+import { HeadingTypeEnum } from '../../Components/Heading/enums/heading-type.enum';
+import Text from '../../Components/Text/Text';
+import { TextHtmlTypeEnum } from '../../Components/Text/enums/text-html-type.enum';
+import { TextTypeEnum } from '../../Components/Text/enums/text-type.enum';
+import styles from './sign-up.module.scss';
+import { SignUpTypes } from './types/sign-up.type';
+
+const SignUp: SignUpTypes = () => {
```
Comment (pailodze) [inline app/(auth)/signUp/page.tsx:16]: SignUpType

File: app/(auth)/signUp/page.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                <input
+                  type="email"
+                  {...register('email', {
+                    required: 'Email is required',
+                  })}
+                  className={`${styles.authorizationInput} ${styles.darkAuthorizationInput}`}
+                  placeholder="Enter Your Email"
+                />
+                {errors.name && (
+                  <span className={styles.error}>{`incorrect email`}</span>
```
Comment (pailodze) [inline app/(auth)/signUp/page.tsx:76]: maybe make error component form forms?

File: app/(auth)/signUp/page.tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
+'use client';
+import axios from 'axios';
```
Comment (pailodze) [inline app/(auth)/signUp/page.tsx:2]: name page signup

File: app/(auth)/signUp/types/sign-up.type.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export type SignUpTypes = () => JSX.Element;
```
Comment (pailodze) [inline app/(auth)/signUp/types/sign-up.type.ts:1]: remove this type i found an alternative

File: app/(auth)/signUp/page.tsx (hunk 6 above)
Comment (pailodze) [inline app/(auth)/signUp/page.tsx:16]: alternative of type is const SignUp: FC

FC means functional component

File: app/(authorized)/global.module.scss [hunk 10]
```diff
@@ -0,0 +1,11 @@
+@import '../Colors/common-colors.scss';
+@import '../Colors/dark-colors.scss';
+@import '../Colors/light-colors.scss';
+
+.container {
+  background-color: $background-color-light-mode;
+  border-radius: 12px;
+}
```
Comment (pailodze) [inline app/(authorized)/global.module.scss:8]: add enter

File: app/(authorized)/layout.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+export default function RootLayout({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>): JSX.Element {
+  return (
+    <html lang="en">
+      <body>
+        <RecoilWrapper>
```
Comment (pailodze) [inline app/(authorized)/layout.tsx:13]: add this recoil wrapper in main layout not in authorized layout

File: app/(authorized)/mainPage/page.module.scss [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+
+.container {
+  display: flex;
+  flex-flow: row wrap;
+  width: 100%;
+  padding: 24px;
+  justify-content: center;
+  background-color: $primary-white;
+}
```
Comment (pailodze) [inline app/(authorized)/mainPage/page.module.scss:13]: add enters

File: app/(authorized)/mainPage/page.tsx [hunk 13]
```diff
@@ -0,0 +1,88 @@
+'use client';
```
Comment (pailodze) [inline app/(authorized)/mainPage/page.tsx:1]: main page has not name!

File: app/(authorized)/page.tsx [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import MainPage from './mainPage/page';
+import styles from './page.module.scss';
+
+export default function Home(): JSX.Element {
+  return (
+    <div className={styles.container}>
+      <div className={styles.content}>
+        <MainPage />
```
Comment (pailodze) [inline app/(authorized)/page.tsx:8]: what is this?

File: app/Components/AuthHeader/AuthHeader.module.scss [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+.content {
+  display: flex;
+  flex-flow: row wrap;
+  width: 100%;
+  justify-content: space-between;
+  background-color: #000;
+  padding: 24px;
+  background-color: $background-color-light-mode;
+}
```
Comment (pailodze) [inline app/Components/AuthHeader/AuthHeader.module.scss:13]: add enter

File: app/Components/AuthHeader/AuthHeader.tsx [hunk 16]
```diff
@@ -0,0 +1,16 @@
+import Logo from '../Header/Logo/Logo';
+import ModeSwitcher from '../Header/ModeSwitcher/ModeSwitcher';
+import styles from './AuthHeader.module.scss';
+
+const AuthHeader = (): JSX.Element => {
+  return (
+    <div className="container">
```
Comment (pailodze) [inline app/Components/AuthHeader/AuthHeader.tsx:7]: styles ?

File: app/Components/Header/ModeSwitcher/ModeSwitcher.module.scss [hunk 17]
```diff
@@ -13,6 +13,9 @@
   width: 48px;
   height: 48px;
 }
```
Comment (pailodze) [inline app/Components/Header/ModeSwitcher/ModeSwitcher.module.scss:15]: add enter

File: app/layout.tsx [hunk 18]
```diff
@@ -25,7 +25,9 @@ export default function RootLayout({
     <html lang="en">
       <body className={inter.className}>
         <RecoilWrapper>
-          <Header />
+          <div className={`${'content'} ${'darkContent'}`}>
```
Comment (pailodze) [inline app/layout.tsx:28]: why do you use template literal?

File: app/page.tsx [hunk 19]
```diff
@@ -1,88 +1,11 @@
...
-import Text from './Components/Text/Text';
-import { TextHtmlTypeEnum } from './Components/Text/enums/text-html-type.enum';
-import { TextTypeEnum } from './Components/Text/enums/text-type.enum';
-import styles from './page.module.scss';
+import Authorization from './(auth)/logIn/page';
+// import MainPage from './mainPage/page';
 
 export default function Home(): JSX.Element {
   return (
-    <div className={`${styles.container} ${styles.lightContainer}`}>
```
Comment (pailodze) [inline app/page.tsx:20]: this should me main page!

**Review by pailodze — CHANGES_REQUESTED** [review]

File: app/Components/AuthHeader/AuthHeader.tsx (hunk 16 above)
Comment (Keretcha) [reply app/Components/AuthHeader/AuthHeader.tsx:7]: this container is from global.css

File: app/(auth)/signUp/types/sign-up.type.ts (hunk 9 above)
Comment (Keretcha) [reply app/(auth)/signUp/types/sign-up.type.ts:1]: ? whichone?

File: app/(auth)/signUp/page.tsx (hunk 6 above)
Comment (Keretcha) [reply app/(auth)/signUp/page.tsx:16]: <3 wow

File: app/layout.tsx (hunk 18 above)
Comment (Keretcha) [reply app/layout.tsx:28]: i tried to make darkmode styles

File: app/Components/LoginForm/LoginForm.tsx [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const router: AppRouterInstance = useRouter();
+
+  const onSubmit = async (values: FieldValues): Promise<void> => {
+    try {
+      const response: AxiosResponse = await axios.post(
+        'http://10.10.51.20:3000/auth/login',
```
Comment (lursmana1) [inline app/Components/LoginForm/LoginForm.tsx:30]: switch

File: app/Components/LoginForm/LoginForm.tsx [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+        values,
+      );
+      const { accessToken } = response.data;
+      if (accessToken) {
+        localStorage.setItem('accessToken', accessToken);
+        console.log('User logged in successfully');
+        router.push('/(authorized)');
+      } else {
+        alert('sworad were!!!');
```
Comment (lursmana1) [inline app/Components/LoginForm/LoginForm.tsx:39]: normal response

File: app/Components/LoginForm/LoginForm.tsx [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+    try {
+      const response: AxiosResponse = await axios.post(
+        'http://10.10.51.20:3000/auth/login',
+        values,
+      );
+      const { accessToken } = response.data;
+      if (accessToken) {
+        localStorage.setItem('accessToken', accessToken);
+        console.log('User logged in successfully');
+        router.push('/(authorized)');
```
Comment (lursmana1) [inline app/Components/LoginForm/LoginForm.tsx:37]: / is enough

File: app/Components/LoginForm/LoginForm.tsx [hunk 23]
```diff
@@ -0,0 +1,131 @@
...
+  const onSubmit = async (values: FieldValues): Promise<void> => {
+    try {
+      const response: AxiosResponse = await axios.post(
+        'http://10.10.51.20:3000/auth/login',
+        values,
+      );
+      const { accessToken } = response.data;
+      if (accessToken) {
+        localStorage.setItem('accessToken', accessToken);
+        console.log('User logged in successfully');
```
Comment (lursmana1) [inline app/Components/LoginForm/LoginForm.tsx:36]: remove this log

File: app/Components/LoginForm/LoginForm.tsx [hunk 24]
```diff
@@ -0,0 +1,131 @@
...
+        router.push('/(authorized)');
+      } else {
+        alert('sworad were!!!');
+      }
+    } catch (err) {
+      console.error('Login failed', err);
```
Comment (lursmana1) [inline app/Components/LoginForm/LoginForm.tsx:42]: alert

File: app/Components/SignupForm/Signup.tsx [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+  const password: string = watch('password');
+  const confirmPassword: string = watch('confirmPassword');
+
+  const onSubmit = async (values: object): Promise<void> => {
+    if (password !== confirmPassword) {
+      setError('password', {});
+      return;
+    }
+
+    try {
```
Comment (lursmana1) [inline app/Components/SignupForm/Signup.tsx:34]: consoles

File: app/Components/SignupForm/Signup.tsx [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+      await axios.post('http://10.10.51.20:3000/auth/register', values);
```
Comment (lursmana1) [inline app/Components/SignupForm/Signup.tsx:35]: host

**Pull request merged**